allofplos/makedb.py

```diff
@@ -105,19 +108,24 @@ class PLOSArticle(BaseModel):
     word_count = IntegerField()
     JATS_type = ForeignKeyField(JATSType, related_name='jats')
 
+class SubjectsPLOSArticle(BaseModel):
+    subject = ForeignKeyField(Subjects)
+    article = ForeignKeyField(PLOSArticle)
+
 class CoAuthorPLOSArticle(BaseModel):
     corr_author = ForeignKeyField(CorrespondingAuthor)
     article = ForeignKeyField(PLOSArticle)
 
 db.connect()
-db.create_tables([Journal, PLOSArticle, ArticleType,
-                      CoAuthorPLOSArticle, CorrespondingAuthor,
-                      JATSType, Affiliations, Country])
+db.create_tables([Journal, PLOSArticle, ArticleType, CoAuthorPLOSArticle,
+                  CorrespondingAuthor, JATSType, Affiliations, Country,
+                  SubjectsPLOSArticle, Subjects])
 
 if args.starter:
-    allfiles = os.listdir('starter_corpus')
+    allfiles = glob.glob(os.path.join(os.path.dirname(__file__),
+                          'starter_corpus','*.xml'))
 else:
-    allfiles = os.listdir(corpusdir)
+    allfiles = glob.glob(os.path.join(corpusdir, '*.xml'))
```

Comment:
Instead of `glob`, you should find that the `Corpus` class will serve this purpose, and you can pass in the directory if it's not the corpus directory. Specifically, `Corpus().iter_files()` is a generator will let you iterate through every article without needing to generate the list of files beforehand.

Reply:
I am using glob because I want to get only xml files, and this directory has a __init__.py file I would grab if I request all of them. Will check iter_files() method.